src/commands/data/query.ts

```diff
@@ -145,24 +143,38 @@ export class DataSoqlQueryCommand extends SfCommand<unknown> {
     timeout: Duration,
     allRows: boolean | undefined = false
   ): Promise<SoqlQueryResult> {
-    connection.bulk2.pollTimeout = timeout.milliseconds ?? Duration.minutes(5).milliseconds;
-    let res: Record[];
+    if (timeout.milliseconds === 0) {
+      const job = new QueryJobV2(
```

Comment:
this would sometimes have results if the query finished quickly.  This way it does what the user asked (open and don't be checking) and returns sooner.